src/libstd/io/mod.rs

```diff
@@ -589,6 +1050,23 @@ pub trait BufRead: Read {
     /// return an error if the read bytes are not valid UTF-8. If an I/O error
     /// is encountered then `buf` may contain some bytes already read in the
     /// event that all data read so far was valid UTF-8.
+    ///
+    /// # Examples
+    ///
+    /// A locked standard input implements `BufRead`:
+    ///
+    /// ```
+    /// use std::io;
+    /// use std::io::BufRead;
+    ///
+    /// let stdin = io::stdin();
+    /// let mut stdin = stdin.lock();
+    /// let mut buffer = String::new();
+    ///
+    /// stdin.read_line(&mut buffer).unwrap();
```

Comment:
I've seen a common footgun with this function where you read lines in a loop but forget to truncate at the top of the loop, perhaps this example could be the standard "read all lines of input from stdin", but explicitly call out:

* The `lines` iterator is possibly preferred
* The call to `truncate` clears state between iterations